### PR TITLE
[#10144] Fill example box for "create a comment for a response" section

### PR DESCRIPTION
**Pull request by Derek-Hardy**
Part of #10144 

Screenshot 2020-06-02 at 12 02 54

**Outline of Solution**
- replace the example box with `comment-edit-form`
- update view dropdown button (point 2) & add comment button (point 3) to be consistent with latest UI

### CODE REVIEW

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts [hunk 1]
```diff
@@ -66,6 +68,18 @@ export class InstructorHelpSessionsSectionComponent extends InstructorHelpSectio
     hasEmailSettingsPanelExpanded: true,
   };
 
+  readonly exampleResponse: ResponseOutput = {
+    responseId: '',
+    giver: 'Alice',
+    giverTeam: 'team 1',
```
Comment (t-cheepeng) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts:74]: Should probably capitalize to `Team 1` to be consistent with rest of examples

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts [hunk 2]
```diff
@@ -66,6 +68,18 @@ export class InstructorHelpSessionsSectionComponent extends InstructorHelpSectio
...
+    giverSection: '',
+    recipient: 'Bob',
+    recipientTeam: 'team 2',
```
Comment (t-cheepeng) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts:77]: same

**Comment by t-cheepeng** [comment]
LGTM

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 3]
```diff
@@ -415,16 +415,18 @@ <h3 *ngIf="displaySubsection(showQuestion, 7, 10)">Adding Comments to Responses<
...
-              Click the <i class="fas fa-comment-alt"></i> icon on the right-hand side of the response you would like to comment on
+              Click the <button type="button" class="btn btn-secondary btn-sm"><i class="fas fa-comment"></i>&nbsp;<i class="fas fa-plus"></i></button> icon on the right-hand side of the response you would like to comment on
             </li>
             <li>
               Fill in the form, which will appear similar to the example below
             </li>
           </ol>
-          <p>EXAMPLE BOX</p>
+          <tm-example-box>
+            <tm-comment-edit-form [response]="exampleResponse"></tm-comment-edit-form>
```
Comment (xpdavid) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:428]: I don't think we should rely on the default fields.

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts [hunk 4]
```diff
@@ -66,6 +68,18 @@ export class InstructorHelpSessionsSectionComponent extends InstructorHelpSectio
...
 
+  readonly exampleResponse: ResponseOutput = {
+    responseId: '',
+    giver: 'Alice',
+    giverTeam: 'Team A',
+    giverSection: 'Section A',
+    recipient: 'Bob',
+    recipientTeam: 'Team B',
+    recipientSection: 'Section B',
+    responseDetails: { questionType: FeedbackQuestionType.TEXT },
```
Comment (xpdavid) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts:79]: Consider using `DEFAULT_CONTRIBUTION_RESPONSE_DETAILS()`

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 5]
```diff
@@ -415,16 +415,30 @@ <h3 *ngIf="displaySubsection(showQuestion, 7, 10)">Adding Comments to Responses<
...
-          <p>EXAMPLE BOX</p>
+          <tm-example-box>
+            <div class="card-body">
+              <tm-comment-edit-form
+                [mode]="CommentRowMode.ADD"
+                [model]="exampleModel"
+                [isFeedbackParticipantComment]="false"
+                [response]="exampleResponse"
+                [questionShowResponsesTo]="[]"
+                [isDisabled]="false"
```
Comment (xpdavid) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:435]: If we want the form to be editable, let's make it editable by

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 6]
```diff
@@ -415,16 +415,30 @@ <h3 *ngIf="displaySubsection(showQuestion, 7, 10)">Adding Comments to Responses<
...
             <li>
               Fill in the form, which will appear similar to the example below
             </li>
           </ol>
-          <p>EXAMPLE BOX</p>
+          <tm-example-box>
+            <div class="card-body">
+              <tm-comment-edit-form
+                [mode]="CommentRowMode.ADD"
+                [model]="exampleModel"
```
Comment (xpdavid) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:431]:
```suggestion
                [(model)]="exampleModel"
```

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts [hunk 7]
```diff
@@ -66,6 +71,26 @@ export class InstructorHelpSessionsSectionComponent extends InstructorHelpSectio
     hasEmailSettingsPanelExpanded: true,
   };
 
+  readonly exampleModel: CommentEditFormModel = {
```
Comment (xpdavid) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts:74]: and delete `readonly`.

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html (hunk 6 above)
Comment (xpdavid) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:431]: This is a nice syntax sugar called BANANA syntax.

https://medium.com/@ofirrifo/custom-angular-component-using-two-way-data-binding-aka-banana-in-a-box-syntax-9eb06b8cfb09

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts (hunk 7 above)
Comment (Derek-Hardy) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts]: Updated, but there did not seem to be changes reflected on edit form.

Comment (xpdavid) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts]: If we do not enabling the 2-way data binding, you will not be able to enter the comment text in the form.

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts [hunk 8]
```diff
@@ -66,6 +71,26 @@ export class InstructorHelpSessionsSectionComponent extends InstructorHelpSectio
     hasEmailSettingsPanelExpanded: true,
   };
 
+  exampleModel: CommentEditFormModel = {
```
Comment (xpdavid) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.ts:74]:
```suggestion
  exampleCommentEditFormModel: CommentEditFormModel = {
```

**Pull request merged**